src/Pimple/PimpleContainerAdapter.php

```diff
@@ -68,17 +68,22 @@ private function addServiceToContainer(ServiceDefinition $definition)
         $this->container[$definition->getName()] = $factory;
     }
 
+    /**
+     * @param array $arguments
+     *
+     * @return array
+     */
     private function resolveArguments(array $arguments)
     {
-        return array_map(
-            function ($argument) {
-                if (isset($this->container[$argument])) {
-                    return $this->container[$argument];
-                }
-
-                return $argument;
-            },
-            $arguments
-        );
+        return array_map([$this, 'resolveArgument'], $arguments);
```

Comment:
@tomphp 

Ha, noticed this library when you tweeted about it some time ago. Not entirely sure, as it makes it a bit difficult renaming methods, but I'll leave it up to you.